backend/src/api/query.rs

```diff
@@ -80,11 +80,20 @@ impl Query {
     fn upload_jwt(context: &Context) -> ApiResult<String> {
         context.require_upload_permission()?;
         match &context.auth {
-            AuthContext::User(data) => Ok(context.jwt.new_upload_token(data)),
+            AuthContext::User(data) => Ok(context.jwt.new_token(data)),
             _ => unreachable!("user not logged in, but has upload permissions"),
         }
     }
 
+    /// Returns a new JWT that can be used to authenticate against Opencast when linking to services
+    /// like Studio and the Editor.
+    fn external_link_jwt(context: &Context) -> ApiResult<String> {
+        match &context.auth {
+            AuthContext::User(data) => Ok(context.jwt.new_token(data)),
+            _ => Err(not_authorized!("trying to get link token without user session")),
+        }
+    }
```

Comment:
Well that would require two fields then, since this API is used to get tokens for both links. :thinking: So I would vote for leaving it like this this way? It's not like this grants users additional privileges.

Reply:
We discussed this internally. The result is e7bbde38f17a3b3356627ebb5e59def150d59722.